### PR TITLE
Fixed algorithm, Added function to map starts and ends, writing tests

### PR DESCRIPTION
**Pull request by dustingraves**
The code is in a working state. 
Tests are complete.

npm install
npm test --coverage

node app.js sample/events.js //events{2-4}.js

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -2,112 +2,35 @@
  * Created by dusting on 10/26/15.
  */
 var getFile = require('./getFile');
+var _ = require('underscore');
+var addToGraph = require('./addToGraph').addToGraph;
+var getStartAndEnd = require('./getStartsAndEnds').rootEnd;
+var findAllPaths = require('./findAllPaths').findAllPaths;
 
-var investigation = getFile.processArgs(process);
+var internal = exports.internal = {};
```
Comment (spoon16) [inline app.js]: why do you export the `internal`? are you doing it for testing?

File: addToGraph.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+exports.addToGraph = function(acc, witnessAccount) {
+    internal.previousEvent = null;
+    internal.acc = acc;
+    _.each(witnessAccount, internal.addEvent);
+    return acc;
+
+};
+
+internal.addEvent = function(event) {
```
Comment (spoon16) [inline addToGraph.js]: I don't think you call `internal.addEvent` outside of this module. You should just change this to.

```
//line:10
_.each(witnessAccount, addEvent);

//line:15
function addEvent(event) {
```

File: getStartsAndEnds.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Created by dusting on 11/4/15.
+ */
+var _ = require('underscore');
+
+var internal = exports.internal = {};
```
Comment (spoon16) [inline getStartsAndEnds.js]: What is the algorithmic complexity of this "Find start and end nodes" part of your algorithm? The BIG O.

File: findAllPaths.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Created by dusting on 11/4/15.
+ */
+var _ = require('underscore');
+
+var internal = exports.internal = {};
+
+//Find all paths between start and end
+exports.findAllPaths = function(graph, start, end, start_path){
```
Comment (spoon16) [inline findAllPaths.js]: going to finish my review after dinner

**Comment by spoon16** [comment]
going to finish review this evening, after dinner

File: addToGraph.js (hunk 2 above)
Comment (dustingraves) [inline addToGraph.js]: I was exporting it but marking it internal, so it was easier to test. I wasn't sure how thorough you wanted the tests. If I just test what goes in and what comes out, I can change it.

File: getStartsAndEnds.js (hunk 3 above)
Comment (dustingraves) [inline getStartsAndEnds.js]: O(n^2) which I know is the worst. I suppose I could find another way to tag them.

**Pull request merged**